community/values/src/main/java/org/neo4j/values/storable/TemporalValue.java

```diff
@@ -1397,11 +1400,59 @@ else if ( conflictingMicroSeconds )
         return truncated;
     }
 
+    static <TEMP extends Temporal> TEMP assertValidArgument( Supplier<TEMP> func )
+    {
+        try
+        {
+            return func.get();
+        }
+        catch ( DateTimeException e )
+        {
+            throw new InvalidTemporalArgumentException( e.getMessage(), e );
+        }
+    }
+
+    static <TEMP extends Temporal> TEMP assertValidUnit( Supplier<TEMP> func )
+    {
+        try
+        {
+            return func.get();
+        }
+        catch ( DateTimeException e )
+        {
+            throw new UnsupportedTemporalUnitException( e.getMessage(), e );
+        }
+    }
+
+    static <OFFSET extends ZoneId> OFFSET assertValidZone( Supplier<OFFSET> func )
+    {
+        try
+        {
+            return func.get();
+        }
+        catch ( DateTimeParseException e )
```

Comment:
Would we not expect a `ZoneRulesException` (also) ?

Reply:
Nice catch, it is a typo, should be `DateTimeException` and not `DateTimeParseException`